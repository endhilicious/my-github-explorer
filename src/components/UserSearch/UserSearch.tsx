import React from 'react';
import styles from './UserSearch.module.scss';
import SearchBar from '@/common/SearchBar';

interface UserSearchProps {
  onSearch: (query: string) => void;
}

const UserSearch: React.FC<UserSearchProps> = ({ onSearch }) => {

  return (
    <div className={styles.userSearch}>
      <SearchBar
        placeholder="Search for users..."
        onSearch={onSearch}
      />
    </div>
  );
};

export default UserSearch;
