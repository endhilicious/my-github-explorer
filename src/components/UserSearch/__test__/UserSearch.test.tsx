import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import UserSearch from '../UserSearch';
// import SearchBar from '@/common/SearchBar';

// Mocking SearchBar component
jest.mock('../../../common/SearchBar', () => {
  return jest.fn(({ onSearch }) => (
    <input
      data-testid="search-input"
      placeholder="Search for users..."
      onChange={(e) => onSearch(e.target.value)}
    />
  ));
});

describe('UserSearch Component', () => {
  it('renders UserSearch component correctly', () => {
    const { container } = render(<UserSearch onSearch={() => {}} />);
    const searchBarInput = container.querySelector('input');

    expect(searchBarInput).toBeInTheDocument();
    expect(searchBarInput?.getAttribute('placeholder')).toBe('Search for users...');
  });

  it('triggers search function when typing', () => {
    const mockOnSearch = jest.fn();
    const { getByTestId } = render(<UserSearch onSearch={mockOnSearch} />);
    const searchInput = getByTestId('search-input');

    fireEvent.change(searchInput, { target: { value: 'user123' } });

    expect(mockOnSearch).toHaveBeenCalledWith('user123');
  });

  it('triggers search function when typing', () => {
    const mockOnSearch = jest.fn();
    const { getByTestId } = render(<UserSearch onSearch={mockOnSearch} />);
    const searchInput = getByTestId('search-input');

    fireEvent.change(searchInput, { target: { value: 'user123' } });

    expect(mockOnSearch).toHaveBeenCalledWith('user123');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<UserSearch onSearch={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
