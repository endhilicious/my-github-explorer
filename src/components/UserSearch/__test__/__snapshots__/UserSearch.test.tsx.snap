// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSearch Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="userSearch"
  >
    <input
      data-testid="search-input"
      placeholder="Search for users..."
    />
  </div>
</DocumentFragment>
`;
